### PR TITLE
Field templates

### PR DESCRIPTION
**Pull request by kettanaito**
## Changes
* **Validation:**
  * Field: Add `validated` property responsible for stating that the Field underwent the validation at least once
  * Form: Provide `validate()` method to validate the whole form, skipping already `validated` fields
* **Misc:**
  * Field: Clean up methods and their descriptions
  * Form: Refine handler methods
  * Form: Refine `updateField` and mapping fields to state methods

### CODE REVIEW

File: src/Fields/index.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import Field from './Field';
+import Condition from './Condition';
```
Comment (Vidlec) [inline src/Fields/index.jsx]: Can't resolve './Condition' :(

File: src/FormProvider.jsx [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-const TValidationRules = PropTypes.shape({
-  type: PropTypes.object,
-  name: PropTypes.object
+export const TValidationRules = PropTypes.shape({
+  type: PropTypes.object, // type-specific field validation rules
+  name: PropTypes.object // name-specific field validation rules
 });
 
 export default class FormProvider extends Component {
```
Comment (Vidlec) [inline src/FormProvider.jsx]: Maybe use name `FormContext` because thats what it is :)

Comment (kettanaito) [reply src/FormProvider.jsx]: Usually those things are named ending with `Provider` (`Provider` from Redux, `ApolloProvider` from Apollo). Don't want to break the pattern)

File: src/Fields/index.jsx (hunk 1 above)
Comment (kettanaito) [reply src/Fields/index.jsx]: `Field.Condition` will be removed.

**Comment by kettanaito** [comment]
## Update
* Introduced `valid` and `invalid` field props useful for custom styling
* Ensure form-level validation by field's type and/or name.

**Pull request merged**